### PR TITLE
bugfix: fix `getConfig` throw `ClassCastException` when use JDK proxy

### PR DESCRIPTION
**Pull request by wangliang181230**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
bugfix: fix `getConfig` throw `ClassCastException` when use JDK proxy, and optimize some log.
BUG修复：修复当使用JDK代理时，`getConfig` 方法获取部分配置时抛出 `ClassCastException` 异常的问题，另外顺便优化了几处日志。

> 注：在 PR https://github.com/seata/seata/pull/5234 中，有两处代理将使用JDK代理，因为 `native-image` 对其他代理的支持比较差。

### CODE REVIEW

**Review by slievrly — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 1]
```diff
@@ -61,83 +65,103 @@ public Configuration provide(Configuration originalConfiguration) {
...
+                                    result = null;
+                                }
+                            }
+                        }
+
+                        // 4. Get config defaultValue from the property object
+                        try {
+                            result = getDefaultValueFromPropertyObject(dataId);
+                        } catch (Throwable t) {
+                            LOGGER.error("Get config '{}' default value from the property object failed:", dataId, t);
```
Comment (slievrly) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java:100]: According to the previous configuration priority, the priority of the value object is greater than the default value priority of the parameter

**Review by slievrly — APPROVED** [review]
LGTM

File: seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java (hunk 1 above)
Comment (wangliang181230) [reply seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java:100]: The object is no longer a spring bean, so I think the priority is no longer higher than the arguments.

**Pull request merged**